### PR TITLE
Controller Test

### PR DESCRIPTION
**Pull request by Othniel2471**
In this PR I:
- Wrote tests for `users_controller` and `posts_controller`.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -2,16 +2,38 @@
...
   end
 
   describe 'GET /show' do
-    it 'returns http success' do
-      get '/posts/show'
+    before(:each) do
+      get '/users/3/posts/1'
+    end
+
+    it 'response status correct' do
```
Comment (brainconnect93) [inline spec/requests/posts_spec.rb]: - [ ] Great effort working on project milestones (i.e. `controller-spec`), But currently, I notice some of the test cases are failing in both the `user` and `post` spec. I would kindly suggest you write a correct test case and be sure they are all passing.

Below is the video recording 👇 


https://github.com/Othniel2471/blog_app/assets/78634841/461b90d9-5a6e-4087-9f18-74a6fef0a3a5

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @Othniel2471, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Good Points 👍

- Descriptive Pull Request. ✔️
- Controller-spec implemented ✔️
- `user_spec.rb` & `post_spec.rb` file seen ✔️ 
- Good use of Git flow ✔️

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (Othniel2471) [reply spec/requests/posts_spec.rb]: @brainconnect93  thank you for the review, but all the tests are passing on my end.

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @Othniel2471 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Tests cases passing 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**